src/layered_config_tree/main.py

```diff
@@ -67,10 +71,10 @@ class ConfigNode:
 
     """
 
-    def __init__(self, layers: List[str], name: str):
+    def __init__(self, layers: list[str], name: str):
         self._name = name
         self._layers = layers
-        self._values = {}
+        self._values: dict[str, tuple[Optional[str], ConfigNodeValue]] = {}
```

Comment:
When would the first value of the tuple be None?

Reply:
I don't think I was ever able to track down why this might actually be None (it's the data `source`). I do know that there are a bunch of tests that pass in None. I tried changing all of those to string values but things got complicated so I gave up. I can try again if you agree the source should never be None (seems like it shouldn't eveer be)

Reply:
Discussed and looks correct. Weird that source is first in the tuple, though